src/main/java/com/autotune/analyzer/recommendations/engine/RecommendationEngine.java

```diff
@@ -2170,10 +2170,9 @@ public void fetchMetricsBasedOnProfileAndDatasource(KruizeObject kruizeObject, T
                             setInterval_end_time(Collections.max(containerDataResults.keySet()));    //TODO Temp fix invalid date is set if experiment having two container with different last seen date
 
                     }
-                } else {
+                } else if (null != kruizeObject.getExperimentType() && kruizeObject.getExperimentType().equalsIgnoreCase(AnalyzerConstants.ExperimentTypes.NAMESPACE_EXPERIMENT)){
```

Comment:
As you can see, the current function has too many if and else conditions, which makes it harder to manage. It would be better to split this logic into separate modules — one for handling namespaces and another for containers. Additionally, we can create a shared module that both can use. Let's refactor the code by introducing different functions, such as fetchNamespaceMetrics and fetchContainerMetrics, to handle these responsibilities.

Reply:
Thanks for pointing that out! I've separated the functions as suggested. Now, the logic is divided into distinct modules for handling namespaces and containers,